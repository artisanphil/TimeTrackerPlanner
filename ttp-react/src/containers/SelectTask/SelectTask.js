import React, { Component } from 'react';
import ToggleStartStop from '../ToggleStartStop/ToggleStartStop';
import './SelectTask.css';


//https://www.robinwieruch.de/react-fetching-data/
//https://blog.hellojs.org/fetching-api-data-with-react-js-460fe8bbf8f2
const API = 'https://localhost:5001/api/';

class SelectTask extends Component {

    constructor(props) {
        super(props);
    
        this.state = {
          plannedTasks: [],
          selectedTask: 0
        };
      }
    
    componentDidMount() {
        if(localStorage.getItem('plannedTasks') != null)
        {
            var data = JSON.parse(localStorage.getItem('plannedTasks'));
            this.setState({ plannedTasks: data, selectedTask: data[0]['planid'] });
        }
        else
        {
            this.fetchTasks();
        }
    }  

    fetchTasks()
    {
        fetch(API + 'plannedtasks')
          .then(response => response.json())
          .then(data => this.savePlanedTasks(data));
    }  
    savePlanedTasks(data) {
        localStorage.setItem('plannedTasks', JSON.stringify(data));
        this.setState({ plannedTasks: data, selectedTask: data[0]['planid'] });
    }  
    handleSelectChange = (event) => {
        this.setState({
            selectedTask: event.target.value
        });
    }      
    
    render() {
        const { plannedTasks } = this.state;

        return (
            <div>
                <select onChange={this.handleSelectChange}>
                {plannedTasks.map(task =>
                    <option value={task.planid} key={task.planid}>
                    {task.catname}: {task.taskname} ({task.projectname})
                    </option>
                )}
                </select>
                <ToggleStartStop selectedTask={this.state.selectedTask} />
            </div>
        );    
    }
}

export default SelectTask;