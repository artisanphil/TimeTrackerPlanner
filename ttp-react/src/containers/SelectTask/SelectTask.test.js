import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SelectTask from './SelectTask';

describe('SelectTask', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SelectTask />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <SelectTask />
    );
    
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});