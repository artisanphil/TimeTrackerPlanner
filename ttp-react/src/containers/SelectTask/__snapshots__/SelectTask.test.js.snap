// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTask has a valid snapshot 1`] = `
<div>
  <select
    onChange={[Function]}
  />
  <button
    className="StartStopButton"
    onClick={[Function]}
  >
    Start
  </button>
</div>
`;
